awx/api/serializers.py

```diff
@@ -3354,14 +3384,15 @@ def validate(self, attrs):
             if validation_errors:
                 errors['variables_needed_to_start'] = validation_errors
 
-        extra_cred_kinds = []
-        for cred in data.get('extra_credentials', []):
+        # Prohibit credential assign of the same CredentialType.kind
+        # Note: when multi-vault is supported, we'll have to carve out an
+        # exception to this logic
```

Comment:
What do you think those exceptions are going to look like in the context of this?

Reply:
Well, namely the fact that you'll be able to assign multiple Vault.  This current PR will yell at you if you try it *because* of this code.  For multi-vault support (coming next), we'll need to loosen the restriction here for Vault, and also add a new way to specify `--vault-id` specific passwords at launch time:  https://github.com/ansible/awx/issues/352#issuecomment-337995178